app/services/foreman_ansible/ansible_report_scanner.rb

```diff
@@ -5,7 +5,13 @@ module ForemanAnsible
   # sets the origin of the report to 'Ansible'
   class AnsibleReportScanner
     class << self
-      def add_reporter_data(report, raw); end
+      def add_reporter_data(report, raw)
+        check_mode = raw['check_mode'] == '1'
+        message = Message.find_or_create check_mode
+        source  = Source.find_or_create 'check_mode'
+        log = Log.create(:message_id => message.id, :source_id => source.id, :report => report, :level => :info)
```

Comment:
I am getting errors about log being invalid - the report is not persisted at this point so we do not have a report id to use when saving the log. You could try to add the log data into `raw['logs']` and then let the [core logic](https://github.com/theforeman/foreman/blob/develop/app/services/report_importer.rb#L84-L104) handle the log creation or maybe `report.logs.build` could help.

Reply:
Thanks, I will try to do that.